### ISSUE TITLE
Categories lost when enabling/disabling a news item

### ISSUE THREAD

**Issue by hputzek** [georgringer/news#166]
As a **non-admin** user when enabling/disabling a news record in list view, all categories of the news item are lost.

**To reproduce the issue you can do the following steps:**
All testing can be done in backend.

1. Login as a **non admin** user. This is important.
2.  Go to the news sysfolder.
3.  Switch to list view.
4.  Edit a news record: Add some categories
5.  Save&close the record
6.  In list view deactivate the record
7.  Activate it again
8. Go to edit mode of the record and check the categories.

Oh no, they are all gone!

This is caused by 
htdocs/typo3conf/ext/news/Classes/Hooks/DataHandler.php:134
```php
$fieldArray['categories'] = trim($fieldArray['categories'], ',');
```

If no changes were made to the categories (like when enabling/disabling a news record), the `$fieldArray['categories']` may not be set.
This can be corrected by doing a quick check before attempting to trim the categories:

```php
// check if categories are present in field array before attempting to trim them.
// Otherwise $fieldArray['categories'] is set empty which causes all categories
// of the recored to be deleted.
if(isset($fieldArray['categories'])) {
   $fieldArray['categories'] = trim($fieldArray['categories'], ',');
}
```

**Comment by georgringer** [comment]
Can you do a PR? that would be great!

**Comment by hputzek** [comment]
I just realized this problem was already fixed in more recent versions of news.
I used 3.2.4, it got fixed in 3.2.5

Thanks for your fast reply anyways.

greetings,
Hendrik
